devtools/aesh/src/main/java/io/quarkus/cli/commands/DevModeCommand.java

```diff
@@ -0,0 +1,161 @@
+
+    @Option(name = "jvm-args", shortName = 'j', description = "JVM arguments to the dev mode process")
+    private String jvmArgs;
+
+    @Option(name = "delete-dev-jar", shortName = 'e', hasValue = false, defaultValue = {
+            "true" }, description = "Delete the dev jar after it finishes")
+    private boolean deleteDevJar;
+
+    @Option(name = "prevent-no-verify", shortName = 'p', defaultValue = {
+            "false" }, hasValue = false, description = "This value is intended to be set to true when some generated bytecode is eroneous causing "
+                    + " the JVM to crash when the verify:non option is set (which is on by default).")
+    private boolean preventNoVerify;
+
+    @Option(name = "no-deps", shortName = 'n', hasValue = false, defaultValue = {
+            "false" }, description = "Whether changes in the projects that appear to be dependencies of the project containing the "
+                    + " application to be launched should trigger hot-reload. By default they do.")
+
+    @OptionList(name = "compiler-args", shortName = 'c', description = "Additional parameters to pass to javac when recompiling changed source files.")
+    private List<String> compilerArgs;
+
+    @Argument(description = "Path to the project, if not set it will use the current working directory")
+    private File projectPath;
+
+    private BuildTool buildTool;
+    private File buildFile;
+
+    @Override
+    public CommandResult execute(CommandInvocation invocation) throws CommandException, InterruptedException {
+
+        if (!verifyProjectStatus(invocation))
+            return CommandResult.FAILURE;
+
+        try {
+            devMode(invocation);
+        } catch (InterruptedException i) {
+            invocation.println("Stopping dev-mode");
+        }
+
+        return CommandResult.SUCCESS;
+    }
+
+    private void devMode(CommandInvocation invocation) throws InterruptedException {
+        if (buildTool.equals(BuildTool.MAVEN)) {
+            File wrapper = ExecuteUtil.getMavenWrapper(projectPath.getAbsolutePath());
+            if (wrapper != null) {
+                ExecuteUtil.executeWrapper(invocation, wrapper, "quarkus:dev");
+            } else {
+                ExecuteUtil.executeMaven(projectPath, invocation, "quarkus:dev");
+            }
+
+        }
+        //do gradle
+        else {
+            File wrapper = ExecuteUtil.getGradleWrapper(projectPath.getAbsolutePath());
+            if (wrapper != null) {
+                ExecuteUtil.executeWrapper(invocation, wrapper, "quarkusDev");
+            } else {
+                ExecuteUtil.executeGradle(projectPath, invocation, "quarkusDev");
+            }
+        }
+    }
+
+    private boolean verifyProjectStatus(CommandInvocation invocation) {
+
+        if (projectPath == null)
+            projectPath = new File(System.getProperty("user.dir"));
+
+        buildTool = QuarkusProject.resolveExistingProjectBuildTool(projectPath.toPath());
```

Comment:
Maybe should make this static method part of `BuildTool` (it's my mistake)?

Reply:
Yes, that's probably useful. I can just change that in this commit. Give me a few min (lunch now).